src/benchmarks-julia.jl

```diff
@@ -1,62 +1,97 @@
-#!/usr/bin/env julia
+options = Base.JLOptions()
+
+@assert VERSION >= v"1.0.2"
+@assert options.can_inline == 1
+@assert options.check_bounds == 2
+@assert options.opt_level == 3
 
 using RigidBodyDynamics
-using StaticArrays
+using Random
+using LinearAlgebra
 
 include("models.jl")
 
-function benchmark_julia_rnea(model, log_filename)
-    robot = parse_urdf(Float64, PATH * "$model.urdf")
-    if model != "lwr"
-        for joint in out_joints(root_body(robot), robot)
-            floatingjoint = Joint(string(joint), frame_before(joint), frame_after(joint), QuaternionFloating{Float64}())
-            replace_joint!(robot, joint, floatingjoint)
-        end
+function write_times(log_filename, times)
+    file = open(log_filename, "w") do io
+        println.(Ref(io), times[2 : end]) # skip the first sample, which includes compilation time
     end
+    return nothing
+end
 
-    states = Array{MechanismState}(undef, NBT)
-    for i = 1:NBT
-        states[i] = MechanismState(robot)
-        rand!(states[i])
+function benchmark_julia_rnea(model, log_filename)
+    robot = parse_urdf(PATH * "$model.urdf", floating=model != "lwr")
+    state = MechanismState(robot)
+    result = DynamicsResult(robot)
+    qs = [similar(configuration(state)) for _ = 1 : NBT]
+    vs = [similar(velocity(state)) for _ = 1 : NBT]
+    v̇s = [similar(velocity(state)) for _ = 1 : NBT]
+    τ = similar(velocity(state))
+    times = Vector{Int}(undef, NBT)
+    Random.seed!(1)
+    for i = 1 : NBT
+        rand!(state)
+        qs[i] .= configuration(state)
+        vs[i] .= velocity(state)
+        rand!(v̇s[i])
     end
+    run_rnea_benchmark(state, qs, vs, v̇s, τ, result.jointwrenches, result.accelerations, times)
+    write_times(log_filename, times)
+    return nothing
+end
 
-    file = open(log_filename, "w")
-    for i = 1:NBT
+function run_rnea_benchmark(state, qs, vs, v̇s, τ, jointwrenches, accelerations, times)
```

Comment:
Extract out a separate function so that code will be specialized on the type of `state`. The exact type of `MechanismState` depends on the actual joint types present in the URDF file. This function barrier forces specialization (a common Julia performance tip, see https://docs.julialang.org/en/v1/manual/performance-tips/#kernel-functions-1).